redsky/streamer.py

```diff
@@ -0,0 +1,41 @@
+from copy import deepcopy as dc
+
+
+def db_store(db, fs_data_name_save_map=None):
+    if fs_data_name_save_map is None:
+        fs_data_name_save_map = {}
+
+    def wrap(f):
+        def wrapped_f(*args, **kwargs):
+            gen = f(*args, **kwargs)
+            for name, doc in gen:
+                fs_doc = dc(doc)
+
+                if name == 'descriptor':
+                    # Mutate the doc here to handle filestore
+                    for data_name in fs_data_name_save_map.keys():
+                        fs_doc['data_keys'][data_name].update(
+                            external='FILESTORE:')
+                    fs_doc.update(
+                        filled={k: False for k in fs_data_name_save_map.keys()}
+                    )
+
+                    doc.update(
+                        filled={k: True for k in fs_data_name_save_map.keys()})
+
+                elif name == 'event':
+                    # Mutate the doc here to handle filestore
+                    for data_name, sub_dict in fs_data_name_save_map.items():
+                        fs_doc = sub_dict['saver'](data_name, fs_doc,
```

Comment:
Re-reading your comment I might be more confused than before. The issue that I raised was one of closing instances prematurely. However, you are discussing multiple descriptors. I actually don't think I was correct in my edit, since when we decorate the generator with `db_store` and the Saver instance we load the saver into the generator's name space. So long as we don't close the Saver until the stop document comes down we should have an open Saver, right?

Your point was more about multiple descriptors, but shouldn't we have one Saver per descriptor? This way we can make certain that the data in the events is saved according to the correct descriptor?